### PR TITLE
In Au tests, poll for excretion of log file rather than waiting fixed delay.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

File: tests/gold_tests/h2/httpbin.test.py [hunk 1]
```diff
@@ -124,9 +124,10 @@
...
-
-# Check Logging
+# Wait for log file to appear, then wait one extra second to make sure TS is done writing it.
 test_run = Test.AddTestRun()
-test_run.DelayStart = 10
-test_run.Processes.Default.Command = 'echo "Delay for log flush"'
+test_run.Processes.Default.Command = (
+    os.path.join(Test.Variables.AtsTestToolsDir, 'condwait') + ' -f ' +
+    os.path.join(ts.Variables.LOGDIR, 'access.log') + ' ; sleep 1'
+)
```
Comment (shinrich) [inline tests/gold_tests/h2/httpbin.test.py]: If I'm understanding your condwait shell script correctly, it incorporates the sleeps.  Why do you append an extra sleep at the invocation sites?

Comment (ywkaras) [reply tests/gold_tests/h2/httpbin.test.py]: For an extra safety margin, to ensure that TS has time to finish writing the log file before its contents are checked.

Comment (shinrich) [reply tests/gold_tests/h2/httpbin.test.py]: Why don't you include that in the script then?

Comment (ywkaras) [reply tests/gold_tests/h2/httpbin.test.py]: OK done.

File: tests/tools/condwait [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# A bash script to wait for a condition to be true.
+#
+# usage:
+#  condwait [ MAX-WAIT [ POST-WAIT ] ] CONDITION
+#
+# CONDITION is the ('test' command) condition to wait for.  (It may contain white space.)
+#
+# MAX-WAIT is the maximum number of seconds to wait for the condition.  If it is omitted, it defaults to 60.
+#
+# MAX-WAIT is the number of seconds to wait after the condition is true.  If it is omitted, it defaults to 0.
```
Comment (bryancall) [inline tests/tools/condwait]: s/MAX-WAIT/POST_WAIT/

**Comment by ywkaras** [comment]
[approve ci Clang-Analyzer]

**Comment by ywkaras** [comment]
OK this is ready for re-review.

**Pull request merged**